sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,312 @@
+from nose import SkipTest
+try:
+    # FIXME This is to skip this import in Python 2.6
+    # To be removed when support for py2.6 ceases
+    from collections import Counter
+except ImportError:
+    pass
+from sklearn.utils.testing import assert_equal, assert_array_equal
+
+
+scikit_learn_path = os.getcwd().replace("/sklearn/tests", "")
+docscrape_path = 'doc/sphinxext/numpy_ext/docscrape.py'
+docscrape = imp.load_source('docscrape',
+                            os.path.join(scikit_learn_path, docscrape_path))
+
+
+def f1(non_default, default=1, default_2=None):
+    """A perfect function for testing
+
+    Parameters
+    ----------
+    non_default : A non default argument
+    default : A default argument, default val=1
+    default_2 : Another default argument, default val=None
+    """
+
+
+def f2(non_default, default=1, default_2=5):
+    """A test function with missing parameters in docstring
+
+    Parameters
+    ----------
+    default : type
+    """
+
+
+def f3(non_default, default=1, default_2=None):
+    """A test function with additional parameter in docstring.
+    
+    The default values in argspec initialized to None are not considered.
+
+    Parameters
+    ----------
+
+    non_default : type
+        A parameter with no default initialization.
+
+    default : type
+        A parameter with a default initialization.
+
+    default_2 : type
+        Another paramter with a default initialization.
+
+    additional : type
+        The additional parameter which is not present in argspec.
+    """
+
+
+def f4(non_default, default=1, default_2=None, *args):
+    """A test function just an additional parameter in docstring.
+    
+    The default values in argspec initialized to None are not considered.
+
+    Parameters
+    ----------
+    foo : type
+    """
+
+
+def f5(non_default, default=1, *args, **kwargs):
+    """A test function with both additional parameter in docstring and argspec.
+    
+    Presence of kwargs makes the additional parameter(s) to not show up
+    as mismatch. Only `default` is shown as missing from docs.
+
+    Parameters
+    ----------
+
+    non_default : type
+        A parameter with no default initialization.
+
+    default_2 : type
+        Another paramter with a default initialization.
+
+    additional : type
+        The additional parameter which is not present in argspec.
+    """
+
+
+def f6(non_default, default=1, default_2=None, *args):
+    """A test function just an additional parameter in docstring.
+    
+    The default values in argspec initialized to None are not considered.
+
+    Parameter
+    ---------
+    foo : type
+    """
+
+
+def get_name(func):
+    parts = []
+    module = inspect.getmodule(func)
+    if module:
+        parts.append(module.__name__)
+    if hasattr(func, 'im_class'):
+        parts.append(func.im_class.__name__)
+    parts.append(func.__name__)
+    return '.'.join(parts)
+
+
+def check_parameters_match(func, doc=None, debug=False):
```

Comment:
If not in the numpydoc project, perhaps this belongs in `sklearn.utils.estimator_checks`